src/SplitProvider.tsx

```diff
@@ -33,74 +25,75 @@ export const SplitContext: SplitReactContext = createContext<
  *    <App />
  *  </SplitProvider>
  */
-const SplitProvider = class extends React.Component<
-  ISplitProviderProps,
-  ISplitContextValues
-> {
-  constructor(props: ISplitProviderProps) {
-    super(props);
+const SplitProvider = ({ config, children }: ISplitProviderProps) => {
+  const {
+    startup = {},
+    scheduler = {},
+    core,
+    features = {},
+    storage = {},
+    debug,
+    impressionListener,
+  } = config;
+  const clientDeps = [
+    startup.readyTimeout,
+    startup.requestTimeoutBeforeReady,
+    startup.retriesOnFailureBeforeReady,
+    startup.eventsFirstPushWindow,
+    scheduler.featuresRefreshRate,
+    scheduler.impressionsRefreshRate,
+    scheduler.metricsRefreshRate,
+    scheduler.segmentsRefreshRate,
+    scheduler.eventsPushRate,
+    scheduler.eventsQueueSize,
+    scheduler.offlineRefreshRate,
+    core.authorizationKey,
+    core.key,
+    core.trafficType,
+    core.labelsEnabled,
+    ...Object.entries(features).map((k, v) => `${k}::${v}`),
+    storage.type,
+    storage.prefix,
+    debug,
+    impressionListener,
+  ];
 
-    /**
-     * Instatiating a factory in order to create a client.
-     * @see {@link https://help.split.io/hc/en-us/articles/360020448791-JavaScript-SDK#2-instantiate-the-sdk-and-create-a-new-split-client}
-     */
-    const factory: SplitIO.ISDK = SplitFactory(props.config);
+  const [client, setClient] = useState<SplitIO.IClient | null>(null);
+  const [isReady, setReady] = useState(false);
+  const [lastUpdate, setUpdated] = useState(0);
 
-    this.state = {
-      client: factory.client(),
-      isReady: false,
-      lastUpdate: null,
-    };
-  }
-
-  /**
-   * Listening for split events
-   */
-  componentDidMount() {
-    const { client } = this.state;
-
-    /**
-     * When SDK is ready this isReady to true
-     */
-    client.on(client.Event.SDK_READY, () => this.setState({ isReady: true }));
-
-    /**
-     * When an update occurs update lastUpdate, this will force a re-render
-     */
-    client.on(client.Event.SDK_UPDATE, () =>
-      this.setState({
-        lastUpdate: Date.now(),
-      }),
-    );
+  useEffect(() => {
+    /** @link https://help.split.io/hc/en-us/articles/360020448791-JavaScript-SDK#2-instantiate-the-sdk-and-create-a-new-split-client */
+    const nextClient = SplitFactory(config).client();
+    setClient(nextClient);
+    nextClient.on(nextClient.Event.SDK_READY, () => {
+      setReady(true);
+      setUpdated(Date.now());
```

Comment:
I just got a random error about setting state while the component was not mounted. I think we need to check that the component is still mounted before setting state since this is an async call that could happen at random times. I'll look into this.

Reply:
Ok.

Reply:
Have you managed to fix the error?

Reply:
I haven't ran into it again. It has to do with HMR because that's the only time that the component is unmounted. I had stubbed out a potential fix that used [`react-use's useMountedState`](https://github.com/streamich/react-use/blob/master/docs/useMountedState.md) to check if the component was mounted before calling the setters.